crates/worker/src/registrar.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt;
+use std::marker::PhantomData;
+
+use crate::codec::{Bincode, Codec};
+use crate::lifecycle::WorkerLifecycleEvent;
+use crate::messages::{FromWorker, ToWorker};
+use crate::native_worker::{DedicatedWorker, NativeWorkerExt, WorkerSelf};
+use crate::scope::WorkerScope;
+use crate::traits::Worker;
+
+/// A Worker Registrar.
+pub struct WorkerRegistrar<W, CODEC = Bincode>
+where
+    W: Worker,
+    CODEC: Codec,
+{
+    _marker: PhantomData<(W, CODEC)>,
+}
+
+impl<W: Worker> fmt::Debug for WorkerRegistrar<W> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("WorkerRegistrar<_>")
+    }
+}
+
+impl<W, CODEC> WorkerRegistrar<W, CODEC>
+where
+    W: Worker,
+    CODEC: Codec,
+{
+    pub(crate) fn new() -> Self {
+        Self {
+            _marker: PhantomData,
+        }
+    }
+
+    /// Sets a new message encoding.
+    pub fn encoding<C>(&self) -> WorkerRegistrar<W, C>
+    where
+        C: Codec,
+    {
+        WorkerRegistrar {
+            _marker: PhantomData,
+        }
+    }
```

Comment:
This seems confusing. Why not just call `new` again? It's not like there's any state that needs to be kept. They both seem to be doing the same thing.